pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -2246,6 +2246,19 @@ public void testFlattenAndOr() {
       }
     }
 
```

Comment:
We may also want to add a test where the boolean expression is as expected (e.g `WHERE col1 = true`) and therefore rewrite is a NO-OP  or a mix of expressions where some are rewritten and some are NO-OP?

Reply:
Added the following tests. WHERE clause containing:
1. Only boolean field.
2. Only boolean field (and) boolean field with equals.
3. Only scalar field.
4. Only scalar field (and) scalar field with equals.
5. Only scalar field (and) scalar field with equals and only boolean field (and) boolean field with equals